guide/src/migration.md

```diff
@@ -234,7 +237,15 @@ impl PyClassAsyncIter {
 
 ### Migrating from the GIL-Refs API to `Bound<T>`
 
-TODO
+To minimise breakage of code using the GIL-Refs API, the `Bound<T>` smart pointer has been introduced by adding complements to all functions which accept or return GIL Refs. This allows code to migrate by replacing the deprecated APIs with the new ones.
+
+For example, the following APIs have gained updated variants:
+- `PyList::new`, `PyTyple::new` and similar constructors have replacements `PyList::new_bound`, `PyTuple::new_bound` etc.
+
+Because the new `Bound<T>` API brings ownership out of the PyO3 framework and into user code, there are a few places where user code is expected to need to adjust while switching to the new API:
+- Code will need to add the occasional `&` to borrow the new smart pointer as `&Bound<T>` to pass these types around (or use `.clone()` at the cost of increasing the Python reference count)
```

Comment:
I think we should qualify "cost" here. Maybe "small cost"? I'd say that generally there is little reason to avoid these `clone()` calls in application unless profiling indicates a bottleneck. (Of course, this reasoning does not apply PyO3 itself where we do not know the usage patterns upfront.)

I would even go so far as explicitly stating that cloning `Bound` is cheap because it just bumps the Python reference count instead of actually cloning the value. (Probably in the inline docs, not in the migration guide.)

Reply:
That's a great point! I don't want to deter folks from using `clone` as lots of reference count changes will be inevitable anyway. I think I'll use "very small" even.

> I would even go so far as explicitly stating that cloning `Bound` is cheap because it just bumps the Python reference count instead of actually cloning the value. (Probably in the inline docs, not in the migration guide.)

Agreed, I think there's a new bit of documentation for `Bound` needed in `types.md`, I've got an item to do that in the tracking issue, and I've added a sub-bullet to remember to mention cloning.